### DISCUSSION TITLE
Controlling image setting in code (code equivalent to command line -define)

### DISCUSSION THREAD

**Discussion by oren-boop** [dlemstra/Magick.NET#838] [Help] [answered]
I'm using ImageMagick.Net to write an image. I'm trying to control some settings in the output image. I can't figure out how to modify these settings from code, although they are available in ImageMagick command line (ImageMagick defines).

I'm specifically interested in modifying the "predictor" setting which is by default 3 when the image is compressed using the "Deflate" method.

Sample code:

        var collection = new MagickImageCollection(args[0]);
        IWriteDefines tiffWriteDefines= new TiffWriteDefines()
        {
            PreserveCompression = true
        };
        collection.Write(@"d:\outImage.tiff", tiffWriteDefines);

These settings are controlled from command line by specifying the

Magick image.tif -define tiff:predictor=1

Thanks

**Comment by dlemstra** [comment]
It is probably the easiest to add this to the `TiffWriteDefines` class. A pull request to add support for this would be helpful for me. And if you take a look at the code of `TiffWriteDefines` you will probably also find the answer to your question.

**Comment by oren-boop** [comment, marked as answer]
I found the answer: to modify this setting you should go over each of the images in the collection and:

foreach (var image in imageCollection)
{
    image.Settings.SetDefine(MagickFormat.Tiff, "predictor", "1");
}
this solved the problem for me.

**Comment by dlemstra** [comment]
In the next version of Magick.NET the `TiffWriteDefines` will have a new `Predictor` property.
